Base/Types.swift

```diff
@@ -0,0 +1,37 @@
+//
```

Comment:
FYI: This is a workaround, as unfortunately, as of Swift 4, you cannot extend an Array where the Element itself is generic without specifying a type as the generic parameter. There's a generics manifesto that specifies this may be coming in future versions and would mean we could do for example:

extension Array<Value> where Element: Weak<Value>, Value: Equatable {

}